crates/utils/src/sensitive.rs

```diff
@@ -0,0 +1,98 @@
+use std::{
+  borrow::Borrow,
+  ops::{Deref, DerefMut},
+};
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, serde::Deserialize, serde::Serialize)]
+#[serde(transparent)]
+pub struct Sensitive<T>(T);
```

Comment:
Here's Sensitive. It AsRefs, AsMuts, Derefs, DerefMuts, Borrows, Serializes, and Deserializes the same as a String, and AsRef's, AsMut's, Borrows, Serializes and Deserializes generically for any T.

So far it's only used to hide Debug info from strings in the codebase, but it could feasibly be used for ip addresses or other information now

Reply:
Is this intentional? It is still logging the sensitive fields:

```
lemmy_1     | 2021-11-30T15:22:28.000177Z  INFO HTTP request{http.method=GET http.scheme="http" http.host=localhost http.target=/api/v3/ws otel.kind="server" request_id=85eab54b-6fcf-409f-83eb-f177a1710df0 http.status_code=101 otel.status_code="OK"}: lemmy_websocket::routes: Message received: "{\"op\":\"Login\",\"data\":{\"username_or_email\":\"oeanhoetaiesahnte\",\"password\":\"oeanhoetaiesahnte\"}}" from id: 7763522254942341211
lemmy_1     | 2021-11-30T15:22:30.016076Z  INFO HTTP request{http.method=GET http.scheme="http" http.host=localhost http.target=/api/v3/ws otel.kind="server" request_id=85eab54b-6fcf-409f-83eb-f177a1710df0 http.status_code=101 otel.status_code="OK"}: lemmy_websocket::routes: Message received: "{\"op\":\"GetSite\",\"data\":{\"auth\":\"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOjcsImlzcyI6Im15X2RvbWFpbiIsImlhdCI6MTYzODI4NTc0OX0.BsY1dUgxf5FO6k7RgQL5gN7TSKaeM5EAd-gvg0R0lUU\"}}" from id: 7763522254942341211
```

Reply:
That's raw JSON that you're logging there. Nothing I can do to help with that

Reply:
relevant line: https://github.com/LemmyNet/lemmy/blob/ad76c758215c97a2ee49b4b5bbfe411ce538ec13/crates/websocket/src/routes.rs#L118

You're dumping the raw text you receive over the websocket to your logs

Reply:
I might suggest something like this:

```rust
#[derive(serde::Deserialize)]
struct DebugOp {
  op: String
}
if let Ok(debug_op) = serde_json::from_str(&m) {
  info!("Message receive: {} from id: {}", debug_op.op, self.id);
} else {
  info!("Unknown message received: {:?} from id: {}", m, self.id);
}
```

Reply:
So this still needs to be solved before merging the PR?

Reply:
This is not a regression. Lemmy already does this.

I could solve it in this PR if you want me to

Reply:
I'd be fine just removing that logging line. Its pry not necessary, as any errors will show up as `LemmyError`s in the perform functions anyway.